compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -342,8 +342,8 @@ object Names {
 
     override def encode: SimpleName = {
       val dontEncode =
-        length >= 3 &&
-        head == '<' && last == '>' && isIdentifierStart(apply(1))
+        length >= 6 &&
+        head == '<' && (this == StdNames.nme.CONSTRUCTOR || this == StdNames.nme.STATIC_CONSTRUCTOR)
```

Comment:
This can be simplified to just checking if this is CONSTRUCTOR or STATIC_CONSTRUCTOR, the checks for length and head aren't needed anymore.

Reply:
you're right! thanks